ancestrymanager/ancestrymanager.go

```diff
@@ -4,20 +4,120 @@ package ancestrymanager
 import (
 	"context"
 	"fmt"
+	"log"
 	"strings"
 
 	"github.com/pkg/errors"
 	"google.golang.org/api/cloudresourcemanager/v1"
 	"google.golang.org/api/option"
+
+	converter "github.com/GoogleCloudPlatform/terraform-google-conversion/google"
 )
 
 // AncestryManager is the interface that wraps the GetAncestry method.
 type AncestryManager interface {
 	// GetAncestry takes a project name and return a ancestry path
-	GetAncestry(project string) (string, error)
+	GetAncestry(project string, options ...GetAncestryOption) (string, error)
+}
+
+type getAncestryRequest struct {
+	project     string
+	hasResource bool
+	tfData      converter.TerraformResourceData
+	hasAsset    bool
+	asset       converter.Asset
+}
+
+// GetAncestryOption sets options for an ancestry tequest
+type GetAncestryOption func(*getAncestryRequest)
+
+// TerraformResource sets the Terraform resource for a getAncestry request
+func TerraformResource(tfData converter.TerraformResourceData) GetAncestryOption {
```

Comment:
It would be easier for reader to figure this out if the function can be named in the form of `With...` or `For...`, such as `ForTerraformResource`.